packages/dev/core/src/Materials/Textures/videoTexture.ts

```diff
@@ -461,6 +461,9 @@ export class VideoTexture extends Texture {
                     });
                 }
                 videoTexture.onDisposeObservable.addOnce(() => {
+                    stream.getTracks().forEach((track) => {

```

Comment:
The WebAPI doesn't make it super clear, but from my understanding of the intent. Whoever creates the stream (via `navigator.getUserMedia()`) should be the one to close it.

In this helper function we don't know if the user created the stream and passed it into CreateFromStreamAsync or if VideoTexture created it internally via CreateFromWebCamAsync (which then calls CreateFromStreamAsync). In the former case it should be up to the caller to close the stream, and in the latter case up to VideoTexture.

Reply:
I would propose capturing the VideoTexture returned from `CreateFromStreamAsync` inside of `CreateFromWebCamAsync` and then adding on another onDisposed handler that closes the stream created by `CreateFromWebCamAsync`.

The reason for the subtle difference is that the caller could be using the MediaStream in multiple instances or outside of Babylon as well. Closing the stream closes it across all instances so you should only do it if you're the owner of the stream.